### DISCUSSION TITLE
Docker on Windows warning about poor performance

### DISCUSSION THREAD

**Discussion by sschuberth** [batect/batect#1020] [Q&A] [answered]
When running Batect under Windows (on Git Bash from Git for Windows), Docker Desktop shows a notification with the following text:

> Docker Desktop has detected that you shared a Windows file into a WSL 2 container, which may perform poorly. Click here for more details.

Clicking on the notification leads to [here](https://docs.docker.com/desktop/windows/wsl/).

Since I'm under the belief Batect is trying its best to achieve good performance on all platforms, I wonder whether this notification from Docker is expected / justified. Any insights on that?

**Comment by charleskorn** [comment, marked as answer]
I suspect that's a reference to this paragraph on that page:

>To get the best out of the file system performance when bind-mounting files, we recommend storing source code and other data that is bind-mounted into Linux containers (i.e., with `docker run -v <host-path>:<container-path>`) in the Linux file system, rather than the Windows file system. 

The warning is justified, but I would expect the performance to be roughly the same as than the equivalent situation on macOS when mounting a file or folder into a container from the host. I think they're trying to point out that there is a better option available to you on Windows (no such better option exists on macOS, which is why they wouldn't show a warning like this there).

**Reply by sschuberth** [reply]
> I think they're trying to point out that there is a better option available to you on Windows

So, is there a way Batect could automatically make use of that better option on Windows?

**Reply by charleskorn** [reply]
Not that I'm aware of unfortunately - Batect would need to sync your code into the WSL environment for you, which is what Docker Desktop is doing.
